src/sentry/utils/concurrent.py

```diff
@@ -46,7 +45,10 @@ def run():
 
 
 @functools.total_ordering
-class PriorityTask(collections.namedtuple("PriorityTask", "priority item")):
+class PriorityTask(NamedTuple):
+    priority: int
+    item: tuple[Callable, Hub, FutureBase]
```

Comment:
`Callable` is an incomplete type -- this should have a signature probably?

mypy should probably be running with `--disallow-any-generics`

Reply:
@asottile-sentry This was completely untyped previously – so what we have here is already an improvement – and it is not entirely obvious to me what the type here should be. 

If you prefer, I can explicitly type this callable as `Callable[..., Any]`, but I believe adding more specific typing or changing mypy configuration would be out of scope for this PR.

Reply:
ah but if you add this file to the stronglist it will require that setting -- at the bottom of `pyproject.toml`

Reply:
looking at the usage below this appears to be `Callable[[], R]`

Reply:
OK, I was able to make some changes to type the callable as `Callable[[], T]`. This file now passes with `--disallow-any-generics`; however, we cannot add it to the stronglist as the file still fails `--disallow-untyped-defs` due to untyped functions. I believe we should fix the `--disallow-untyped-defs` separately, since it is not directly related to the `PriorityTask` typing we are trying to improve here.